nifi-nar-bundles/nifi-snmp-bundle/nifi-snmp-processors/src/main/java/org/apache/nifi/snmp/processors/GetSNMP.java

```diff
@@ -134,59 +141,77 @@
             REL_FAILURE
     )));
 
+    private volatile GetSNMPHandler snmpHandler;
+
+    @OnScheduled
+    public void init(final ProcessContext context) throws InitializationException {
+        initSnmpManager(context);
+        snmpHandler = new GetSNMPHandler(snmpResourceHandler);
+    }
+
     @Override
     public void onTrigger(final ProcessContext context, final ProcessSession processSession) {
         final SNMPStrategy snmpStrategy = SNMPStrategy.valueOf(context.getProperty(SNMP_STRATEGY).getValue());
         final String oid = context.getProperty(OID).getValue();
+        final Optional<FlowFile> flowfile = Optional.ofNullable(processSession.get());
+        Map<String, String> flowFileAttributes = flowfile.map(FlowFile::getAttributes).orElse(Collections.emptyMap());
 
         if (SNMPStrategy.GET == snmpStrategy) {
-            performSnmpGet(context, processSession, oid);
+            performSnmpGet(context, processSession, oid, flowFileAttributes);
         } else if (SNMPStrategy.WALK == snmpStrategy) {
-            performSnmpWalk(context, processSession, oid);
+            performSnmpWalk(context, processSession, oid, flowFileAttributes);
         }
     }
 
-    private void performSnmpWalk(final ProcessContext context, final ProcessSession processSession, final String oid) {
+    private void performSnmpWalk(final ProcessContext context, final ProcessSession processSession, final String oid,
+                                 final Map<String, String> flowFileAttributes) {
         try {
-            final SNMPTreeResponse response = snmpRequestHandler.walk(oid);
+            final SNMPTreeResponse response = snmpHandler.walk(oid, flowFileAttributes);
             response.logErrors(getLogger());
             FlowFile flowFile = createFlowFileWithTreeEventProperties(response, processSession);
             processSession.getProvenanceReporter().receive(flowFile, response.getTargetAddress() + "/" + oid);
             processSession.transfer(flowFile, REL_SUCCESS);
         } catch (SNMPWalkException e) {
             getLogger().error(e.getMessage());
             context.yield();
-            processSession.rollback();
         }
     }
 
-    private void performSnmpGet(final ProcessContext context, final ProcessSession processSession, final String oid) {
+    private void performSnmpGet(final ProcessContext context, final ProcessSession processSession, final String oid,
+                                final Map<String, String> flowFileAttributes) {
         final SNMPSingleResponse response;
         try {
-            response = snmpRequestHandler.get(oid);
+            response = snmpHandler.get(oid, flowFileAttributes);
             final FlowFile flowFile = processSession.create();
             addAttribute(SNMPUtils.SNMP_PROP_PREFIX + "textualOid", context.getProperty(TEXTUAL_OID).getValue(), flowFile, processSession);
             final String provenanceAddress = response.getTargetAddress() + "/" + oid;
```

Comment:
Is this going to be okay even if we are getting OIDs from the flowfiles and not from the property?
```java
            final String provenanceAddress = response.getTargetAddress() + "/" + oid;
```